### PR TITLE
Acquisition channel

### PR DESCRIPTION
**Pull request by ukutaht**
### Changes

This PR will detect and store the [acquisition channel](https://feedback.plausible.io/175) for each session. The way we detect channels is based on GA4 channels as documented [here](https://support.google.com/analytics/answer/9756891?hl=en).

I've also added rules to detect as paid search when:
* Source is `Google` and `gclid` url parameter is present
* Source is `Bing` and `msclkid` url parameter is present

For now the plan is just to store this data and evaluate whether it looks correct. In a follow-up PR I will add an 'Acquisition Channel` report to the UI and make this data available from the Stats API as well.

I'm planning to extract the DB migration before deploying this PR.

### Tests
- [x] Automated tests have been added

### Changelog
Will add once the feature is visible to users

### Documentation
Will add once the fetaure is visible to users

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: lib/plausible/ingestion/acquisition.ex [hunk 1]
```diff
@@ -24,4 +29,149 @@ defmodule Plausible.Ingestion.Acquisition do
...
+  end
+
+  defp video_source?(nil), do: false
+
+  defp video_source?(source) do
+    @source_categories[String.downcase(source)] == "SOURCE_CATEGORY_VIDEO"
+  end
+
+  defp paid_medium?(medium) do
+    Regex.match?(~r/^(.*cp.*|ppc|retargeting|paid.*)$/, medium || "")
```
Comment (zoldar) [inline lib/plausible/ingestion/acquisition.ex:175]: won't the `cp` pattern be too susceptible to false positives?

File: lib/plausible/ingestion/acquisition.ex [hunk 2]
```diff
@@ -1,5 +1,10 @@
 defmodule Plausible.Ingestion.Acquisition do
   @moduledoc false
+  @source_categories Application.app_dir(:plausible, "priv/ga4-source-categories.csv")
```
Comment (zoldar) [inline lib/plausible/ingestion/acquisition.ex:3]: Would be good to declare `priv/ga4-source-categories.csv` as an external resource https://hexdocs.pm/elixir/Module.html#module-external_resource

Like here, for example: https://github.com/plausible/analytics/blob/master/lib/plausible/stats/json_schema.ex#L9

**Review by zoldar — APPROVED** [review]

**Review by macobo — APPROVED** [review]

File: lib/plausible/ingestion/acquisition.ex (hunk 1 above)
Comment (ukutaht) [reply lib/plausible/ingestion/acquisition.ex:175]: Hmm yeah good point. It's copied verbatim from GA4 documentation:

Screenshot 2024-09-05 at 11 32 41

So I'm inclined to keep it as is

File: lib/plausible/ingestion/acquisition.ex (hunk 2 above)
Comment (ukutaht) [reply lib/plausible/ingestion/acquisition.ex:4]: Makes sense! Done in https://github.com/plausible/analytics/pull/4489/commits/c3d5e7f8c7c5c15a61919a9de6b6557ec401e7e6

**Pull request merged**